python/imageutils.py

```diff
@@ -295,125 +185,55 @@ def reconstruct_block_slice(self, reconstructed, legend, mem):
 
         print y_size, z_size, x_size
 
+        total_read = 0
+        total_seek = 0
+        total_defrag = 0
+        total_write = 0
+
         with open(legend, "r") as legend:
 
             remaining_mem = mem
-            data = None
-            start_x = 0
-            start_z = 0
-            start_y = 0
+            data_dict = {}
 
-            prev_block_start = (0,0,0) 
-            for block_name in legend:
-                
-                block_name = block_name.strip()
-                print "Reading block {0}".format(block_name)
-                block_pos = split_ext(block_name)[0].split('_')
-
-                #we will start our (partial) slice at these block coordinates
-                if remaining_mem == mem:
-                    start_x = int(block_pos[-1])
-                    start_z = int(block_pos[-2])
-                    start_y = int(block_pos[-3])
-                    
-                block = nib.load(block_name)
-                block_shape = block.header.get_data_shape()
-                bytes_per_voxel_block = block.header['bitpix']/8
-                block_bytes = block.header.single_vox_offset + bytes_per_voxel_block*(block_shape[0]*block_shape[1]*block_shape[2])
+            unread_split = None
+
+            for split_name in legend:
+               
+                print "Remaining mem: ", remaining_mem 
+                split_name = split_name.strip()
+
+                if unread_split is not None:
+                    unread_split, remaining_mem, read_time = insert_in_dict(data_dict, unread_split, remaining_mem, bytes_per_voxel, y_size, z_size)
+
+                if unread_split is None:    
```

Comment:
right, I had missed part of this. I guess what I don't like is that the `None` returned by `insert_in_dict`in its first argument is actually meaningful as it means "no memory for this split". Perhaps a `try-catch` would be clearer (see other comment on returning None vs Exception). This is pure style, but I'd say it's important that this part of the code is clear if we don't want to introduce bugs here in the future.